projects/core/src/button/button.element.ts

```diff
@@ -83,34 +83,42 @@ export class CdsButton extends CdsBaseButton {
   @property({ type: String })
   loadingState: keyof typeof ClrLoadingState = ClrLoadingState.default;
 
+  get disabled(): boolean {
+    return super.disabled;
+  }
+
+  set disabled(value: boolean) {
+    // track disabled state set from outside, in order to restore value after button is back in default loading state
+    this._disabled_from_outside = value;
+
+    // do NOT propagate disabled state, unless we're in default loading state
+    // otherwise button can be enabled while being in a non-default loading state, e.g. loading
+    if (this.isDefaultLoadingState(this.loadingState)) {
+      super.disabled = value;
+    }
+  }
+
   firstUpdated(props: PropertyValues<this>) {
     super.firstUpdated(props);
 
     if (!this.isDefaultLoadingState(this.loadingState)) {
-      this.disabled = true;
+      super.disabled = true;
     }
   }
 
   update(props: PropertyValues<this>) {
     if (props.has('loadingState')) {
-      if (this.isDefaultLoadingState(props.get('loadingState') as string)) {
-        // track prior disabled state to set prior value after button is re-enabled from a loading state
-        this._disabled = this.disabled;
-      }
-
-      if (props.get('loadingState') !== undefined) {
-        if (this.isDefaultLoadingState(this.loadingState)) {
-          this.enableButton();
-        } else {
-          this.disableButton();
-        }
+      if (this.isDefaultLoadingState(this.loadingState)) {
```

Comment:
I'm trying to remember why, but I think there was a reason I had to have the undefined check here in the last PR. Otherwise I think disabled gets called if other prop values change - we only want to run this is loadingState is the prop changing

Reply:
I remember why - props *always* had loadingState, but unless it's being changed, it's undefined. I found that without the undefined check, the else statement was being called on the initial render when `loadingState` was set, but the button hadn't been painted to the dom yet so the width was 0px. 

I added a test for this situation.